server/methods/core/shop.js

```diff
@@ -57,6 +58,46 @@ Meteor.methods({
   },
 
   /**
+   * shop/createVendorShop
+   * @param {String} shopAdminUserId - optionally create shop for provided userId
+   * @param {Object} shopData - optionally provide shop object to customize
+   * @return {String} return shopId
+   */
+  "shop/createVendorShop": function(shopVendorUserId) {
+    check(shopVendorUserId, Match.Optional(String));
+    // must have owner access to create new shops
+    if (!Reaction.hasOwnerAccess()) {
+      throw new Meteor.Error(403, "Access Denied");
+    }
+
+    const currentUser = Meteor.userId();
+ 
+    if (!currentUser) {
+      throw new Meteor.Error("Unable to create shop with specified user");
+    }
+
+    // identify a shop admin
+    const userId = shopVendorUserId || Meteor.userId();
+    const adminRoles = Roles.getRolesForUser(currentUser, Reaction.getShopId());
+
+    const shop = createShops();
+
+    shop._id = Random.id();
+    shop.name = 'Shop Name';
```

Comment:
shop.name = 'Shop Name'

If we are creating a shop, shouldn't the name come from the shop admin? or?